core/src/main/scala/com/databricks/labs/remorph/Main.scala

```diff
@@ -36,6 +38,15 @@ object Main extends App with ApplicationContext {
         case nok: KoResult =>
           prettyPrinter(nok)
       }
+
+    case Payload("debug-lineage", args) =>
+      val dst = new File(args("dst"))
+      val queryHistory = folderQueryHistoryProvider(args("src")).history()
```

Comment:
stack this PR on top of https://github.com/databrickslabs/remorph/pull/1051 - there's a fundamental change how we treat the query history.